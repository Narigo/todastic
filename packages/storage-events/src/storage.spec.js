const fs = require("fs");
const storage = require("./storage.js");

describe("storage-events", () => {
  const event1 = { eventType: "ADDED_TODO", eventId: "1", data: { title: "Implement event logging" } };
  const event2 = { eventType: "ADDED_TODO", eventId: "2", data: { title: "Another event added" } };

  it("can save to and load from a file", () => {
    const logger = storage.createLogger();
    const tmpFile = `${__dirname}/__tests__/tmp-file.log`;

    logger.log(event1);
    logger.log(event2);

    fs.writeFileSync(tmpFile, JSON.stringify(logger.getEvents()));
    const contents = fs.readFileSync(tmpFile).toString();
    expect(logger.getEvents()).toEqual(JSON.parse(contents));
  });

  it("replaying from event logger should be the same as a simple list", () => {
    const logger = storage.createLogger();

    logger.log(event1);
    logger.log(event2);

    expect(storage.replay([event1, event2])).toEqual(storage.replay(logger.getEvents()));
  });
});
