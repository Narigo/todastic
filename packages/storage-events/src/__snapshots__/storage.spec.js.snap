// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storage-events ADDED_TODO event can add todos in a more complex example 1`] = `
Array [
  Object {
    "id": 1,
    "title": "Create a parent todo",
  },
  Object {
    "children": Array [
      Object {
        "id": 3,
        "parentId": 2,
        "title": "Create a child in another parent todo",
      },
      Object {
        "id": 5,
        "parentId": 2,
        "title": "Create another child todo in parent 2",
      },
    ],
    "id": 2,
    "title": "Create another parent todo",
  },
  Object {
    "id": 4,
    "title": "Create a third parent todo",
  },
]
`;
